lib/commands/network.js

```diff
@@ -76,13 +76,9 @@ commands.toggleFlightMode = async function () {
   });
 };
 
-commands.setGeoLocation = async function (location) {
-  /*
-   * TODO: Implement isRealDevice(). This method fails on
-   * real devices, it should throw a NotYetImplementedError
-   */
-  let cmd = `geo fix ${location.longitude} ${location.latitude}`;
-  return await this.adb.sendTelnetCommand(cmd);
+commands.setGeoLocation = async function (location) {    
+  let cmd = ['am', 'startservice', '-e', 'longitude', location.longitude, '-e', 'latitude', location.latitude, 'io.appium.settings/.LocationService'];
```

Comment:
this should really be broken up into separate lines according to our style guide so that it's not longer than 80 chars:

```js
let cmd = [foo, bar, baz,  // <-- no more than 80 chars
           abc, qed, xyz];
```